src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -390,21 +381,35 @@ public async Task UpdateBufferContentsAsync(CancellationToken cancellationToken)
                             edit.Replace(startPosition: 0, _textBuffer.CurrentSnapshot.Length, generatedSource.ToString());
                             edit.Apply();
                         }
+
+                        // If the file isn't already open, open it now. We may transition between opening and closing
+                        // if the file is repeatedly appearing and disappearing.
+                        var connectToWorkspace = _workspace.Options.GetOption(EnableOpeningInWorkspace) ?? false;
```

Comment:
when would we *not* want to open a document in a workspace?

Reply:
@CyrusNajmabadi This is just a feature flag because to mitigate risk since we're putting this in late for 16.10. We expect to remove this in 16.11 or 17.0.